### DISCUSSION TITLE
Xyce/LinearSolvers/Amesos/Pardiso support

### DISCUSSION THREAD

**Discussion by AMSSWANGYIJIA** [Xyce/Xyce#79] [Q&A]
Hi everyone ! Recently our lab is performing some very large scale power grids' benchmark tests. One of the experiment is to perform DCOP operations using some direct solvers, at first we tried KLU, however it failed in most of the cases(most of the sizes of test power grids are between 1E7 to 4E7). Therefore we try to use some other options such as pardiso. We noticed that there seems potential pardiso support in N_LAS_AmesosSolver.C. However we do not know the right cmake options to compile the corresponding Trilinos library and link pardiso. We are wondering is there any advice on running Xyce with pardiso?

**Comment by Karlsefni2012** [comment]
Thanks for your interest in Xyce.  We have experimented with using Pardiso, but haven't made it part of our standard build of Xyce.  The version of Pardiso that we have used is the one that can be found in the Intel MKL library.  So, it is necessary to use the Intel compilers to get it.   That version is freely available as long as you have that compiler.

We have had some success using that version, but it has to date mostly been an experiment/demonstration for us.  We do have a internal issue to include it in our release builds but that hasn't happened yet.

I'll also point out that there are more recent versions of Pardiso, according to their website.   That newer version works really well on circuits, according to the results posted there.  However, that version isn't open-source and we currently don't have access to it.  So we have no experience with it.  But having said that, the old version that we have tried sometimes works well for large circuits, and can often do better than KLU.

**Comment by tcaduser** [comment]
The Intel MKL is separate from the Intel Compiler, and both are free under the community license for windows, linux, and macOS.  There is no need to use both the library and the compiler, and they can be installed separately.  
https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl.html#gs.g7fhx5

I have successfully used the Intel MKL with Pardiso on my projects, compiling with gcc, clang, and visual studio.
https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html

**Reply by Karlsefni2012** [reply]
Ah!  I stand corrected.  And this is why I didn't attempt to provide any build instructions for Xyce with Pardiso.  I am not the person who experimented with it; I was mostly repeating what I remember about it.

Possibly that person will chime in here ....
